src/network/networkd-sysctl.c

```diff
@@ -40,10 +42,82 @@ static void manager_set_ip_forwarding(Manager *manager, int family) {
                                   enable_disable(t), af_to_ipv4_ipv6(family));
 }
 
+static void sysctl_overwrite_check(sd_event_source *source, const struct inotify_event *event, Hashmap *sysctls) {
+        _cleanup_free_ char *buf = NULL, *path = NULL;
+        const char *dir_path, *value;
+        size_t len;
+        int r;
+
+        assert(event);
+
+        r = sd_event_source_get_inotify_path(source, &dir_path);
+        if (r < 0)
+                return;
+
+        path = path_join(dir_path, event->name);
+        if (!path)
+                return;
+
+        value = hashmap_get(sysctls, path);
+
+        /* We don't handle this sysctl */
+        if (!value)
+                return;
+
+        if (read_full_file(path, &buf, &len) < 0)
+                return;
+
+        delete_trailing_chars(buf, "\n");
+
+        if (!streq(value, buf))
+                log_warning("sysctl %s was changed from '%s' to '%s'", path, value, buf);
+}
+
+static int sysctl_inotify_manager(sd_event_source *source, const struct inotify_event *event, void *userdata) {
+        Manager *manager = ASSERT_PTR(userdata);
+
+        sysctl_overwrite_check(source, event, manager->sysctls);
+
+        return 0;
+}
+
+static int sysctl_inotify_link(sd_event_source *source, const struct inotify_event *event, void *userdata) {
+        Link *link = ASSERT_PTR(userdata);
+
+        sysctl_overwrite_check(source, event, link->sysctls);
+
+        return 0;
+}
+
+static int manager_inotify_add(Manager *manager) {
+        sd_event_source **source;
+
+        assert(manager);
+
+        source = manager->sysctl_event_sources;
+
+        FOREACH_STRING(p,
+                "/proc/sys/net/ipv4/conf/all",
+                "/proc/sys/net/ipv4/conf/default",
+                "/proc/sys/net/ipv6/conf/all",
+                "/proc/sys/net/ipv6/conf/default") {
+                int r;
+
+                r = sd_event_add_inotify(manager->event, source, p, IN_CLOSE_WRITE | IN_ONLYDIR, sysctl_inotify_manager, manager);
+                if (r < 0)
+                        return log_warning_errno(r, "Failed to watch sysctl %s: %m", p);
```

Comment:
hmm, so we generally want that errors fatal to a function (i.e. where the function returns with an error code right-away) are logged at LOG_ERR log level. But errors that it ignores for code path are logged at LOG_WARN. right now you return here but then you use LOG_WARN, which is the invalid combination.

I think this kind of error is actually the type you want to ignore (who knows, people with ipv6 turned off will have this call fail for the ipv6-specific dirs, and we should just be fine with that). Hence, please drop the `return`, but keep the `log_warning_errno(` in place.

there's one more rule though: errors that we ignore should have their log message suffixed with `, ignoring`. Here this means: `"Failed to watch sysctl '%s', ignoring: %n"`